Datasets/WordSeg/WordSegDataset.swift

```diff
@@ -48,6 +71,11 @@ public struct WordSegDataset {
     return strings
   }
 
+  /// Returns the union of all characters in `training` and `otherSequences`.
+  ///
+  /// - Parameter eos: text to be used as the end of sequence marker.
+  /// - Parameter eow: text to be used as the end of word marker.
+  /// - Parameter pad: text to be used as the padding marker.
```

Comment:
See notes above about "character"

```suggestion
  /// Returns the union of all characters in `training` and `otherSequences`.
  ///
  /// - Parameter eos: the end of sequence marker.
  /// - Parameter eow:the end of word marker.
  /// - Parameter pad: the padding marker.
```

We should talk about what these markers really mean; I think we can do better in the comment.

`otherSequences` needs to be described.  Generally the use of current Swift variadics, especially in a `private` method is questionable; we should talk about that. 

The implementation below looks like a terribly inefficient algorithm also, let's talk about that too.

Reply:
Again, [No raw loops](https://developer.apple.com/videos/play/wwdc2018/223/)